dashboards-reports/public/components/context_menu/context_menu.js

```diff
@@ -126,11 +123,8 @@ const generateInContextReport = async (
 };
 
 // try to match uuid and user entered custom-id followed by '?' in URL, which would be the saved search id for discover URL
-// custom id example: v1s-f00-b4r1-01
-const getUuidFromUrl = () =>
-  window.location.href.match(
-    /([0-9a-z]+-[0-9a-z]+)+\?/
-  );
+// custom id example: v1s-f00-b4r1-01, Filebeat-Apache-Dashboard-ecs,
+const getUuidFromUrl = () => window.location.href.match(/[0-9a-zA-Z-]+\?/);
```

Comment:
not too sure but i think id can be any url-encoded string. couldn't find references, if you also don't know then this should be ok

Reply:
I couldn't  find any reference of saved object id format in Dashboards codebase or docs, I figure it's better we use the same logic. Or I wonder if some services is exposed from Dashboards core for save object id. But for the sake of time, let's use this to handle most use case.